### PR TITLE
frontend: Fix MetadataDisplay to skip ResourceClasses

### PR DESCRIPTION
**Pull request by illume**
Ones that we can't get a routeName from skip.

## How to use

See https://github.com/kinvolk/headlamp/issues/649

## Testing done

...

### CODE REVIEW

File: frontend/src/components/common/Resource/MetadataDisplay.tsx [hunk 1]
```diff
@@ -45,21 +45,35 @@ export function MetadataDisplay(props: MetadataDisplayProps) {
...
-        {i < numItems - 1 && <br />}
-      </>
-    ));
+    return ownerReferences
+      .map((ownerRef, i) => {
+        if (ownerRef.kind in ResourceClasses) {
+          let routeName;
+          try {
+            routeName = new ResourceClasses[ownerRef.kind]().detailsRoute;
+          } catch (e) {
```
Comment (joaquimrocha) [inline frontend/src/components/common/Resource/MetadataDisplay.tsx:54]: An error shouldn't occur here, AFAIU. Or is there something I am missing?

**Review by joaquimrocha — APPROVED** [review]
Looks like a nice change. We do have to think about what we can do for unknown resources like in this case. This is a good solution for the owner ref.
I left a comment about the exception, but if it really makes sense to have it, then it's fine for me.

File: frontend/src/components/common/Resource/MetadataDisplay.tsx (hunk 1 above)
Comment (illume) [reply frontend/src/components/common/Resource/MetadataDisplay.tsx:54]: My guess is `ResourceClasses[ownerRef.kind]` isn't callable, so `new ResourceClasses[ownerRef.kind]()` fails.

Comment (joaquimrocha) [reply frontend/src/components/common/Resource/MetadataDisplay.tsx:54]: Okay. It should be callable as that's what we're setting those to be. But I am fine using it defensively here.

**Pull request merged**